### PR TITLE
fix(SyncGenerator): Generate missing attributes

### LINKED ISSUE #10842: Sync generator should generate attributes
## What would you like to be added:

Almost all classes in UWP/WinUI have attributes declared. While some are not necessarily useful in Uno Platform context, some are, e.g. `[ContentProperty]`.

Sync generator should generate these automatically if they are not present in non-generated code already.

## Why is this needed:

Avoid missing attributes.

## For which Platform:

- [ ] WebAssembly
- [ ] Android
- [ ] iOS
- [ ] macOS (AppKit)
- [ ] Mac Catalyst
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Linux Framebuffer
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #10842

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UWPSyncGenerator/SyncGenerator.cs [hunk 1]
```diff
@@ -124,5 +122,126 @@ private void WriteType(INamedTypeSymbol type, IndentedStringBuilder b)
...
+			if (attributeFullyQualifiedName == "System.AttributeUsageAttribute")
+			{
+				var attributeTargets = GetAttributeTargetsString((AttributeTargets)attribute.ConstructorArguments.Single().Value);
+				// For some reason, attribute.ConstructorArguments.Single().Value can be "zero" for some attributes, e.g, OverridableAttribute.
+				// From documentation: https://learn.microsoft.com/en-us/uwp/api/windows.foundation.metadata.overridableattribute?view=winrt-16299
+				// This is `System.AttributeTargets.InterfaceImpl`, but there is no such member in AttributeTargets, so it's probably not .NET-related.
+				// We ignore adding the attribute for this case.
+				if (attributeTargets.Length > 1)
+				{
+					// TODO: AllowMultiple and Inherited properties.
```
Comment (Youssef1313) [inline src/Uno.UWPSyncGenerator/SyncGenerator.cs:166]: Draft for now until I implement this.

**Comment by Youssef1313** [comment]
@jeromelaban Could you take a look?

**Comment by jeromelaban** [comment]
This one would make sense to move to the breaking changes branch

File: src/Uno.UWPSyncGenerator/AttributeGeneration/BindableAttributeDescription.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	{
+		// This attribute doesn't have any constructor/named arguments.
+		// It was already handled by TryGenerateAttributeParametersCommon in AttributeDescriptionBase.
+		// We should never hit this code path, but if we did, just don't handle the attribute.
+		return null;
+	}
+
+	private protected override bool CanHandle(string fullyQualifiedAttributeName)
+	{
+		return fullyQualifiedAttributeName == "Windows" + ".UI.Xaml.Data.BindableAttribute";
```
Comment (jeromelaban) [inline src/Uno.UWPSyncGenerator/AttributeGeneration/BindableAttributeDescription.cs:19]: MUX is not needed here?

Comment (Youssef1313) [reply src/Uno.UWPSyncGenerator/AttributeGeneration/BindableAttributeDescription.cs:19]: There are no much types with BindableAttribute in WinUI it seems, the one I can see is https://learn.microsoft.com/en-us/windows/winui/api/microsoft.ui.xaml.controls.treeviewnode?view=winui-2.8, and is using `Windows.UI.Xaml.Data.BindableAttribute`. It wouldn't harm though to handle both here.

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**